src/mlir/generator.cc

```diff
@@ -915,6 +955,55 @@ namespace mlir::verona
     }
   }
 
+  llvm::Expected<ReturnValue> Generator::parseFieldRead(const ::ast::Ast& ast)
+  {
+    assert(AST::isMember(ast) && "Bad node");
+
+    // Find the variable to extract from
+    auto ref = AST::getLocalRef(ast);
+    auto var = symbolTable.lookup(ref);
+
+    // Get the field name
+    auto field = AST::getID(ast);
+    auto fieldAttr = StringAttr::get(field, context);
+
+    // Find the field type
+    auto type = extractClassType(var.getType());
```

Comment:
This won't work with unknown

Reply:
Right. I can't "extract" a field from an unknown class type. The dialect constructs require the types to be compatible and the field type to be the same as the extracted type.

I could, obviously, remove that requirement on both `field_read` and `field_write` and leave everything unknown, then this whole type propagation would disappear.

Reply:
That is what I think should be done.  After inference and checking, we will remove unknowns.

Reply:
We'll probably have to do that with every dialect construct, not just a few.

It was my original plan, I was just trying to adapt to the dialect constructs.